catalog/core/catalog-core-directorymonitor/src/test/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitorComponentTest.java

```diff
@@ -0,0 +1,308 @@
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerClass;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+
+import com.google.common.io.Files;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.content.operation.CreateStorageRequest;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.security.service.SecurityManager;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerClass.class)
+public class ContentDirectoryMonitorComponentTest {
+    private static final int TIMEOUT_IN_SECONDS = 10;
+
+    @Inject
+    private BundleContext bundleContext;
+
+    private List<ServiceRegistration> serviceRegistrations = new ArrayList<>();
+
+    @Inject
+    private CamelContext camelContext;
+
+    @Inject
+    private ContentDirectoryMonitor contentDirectoryMonitor;
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private String directoryPath;
+
+    private CatalogFramework catalogFramework;
+
+    @Configuration
+    public Option[] config() throws IOException {
+
+        return options(testDependencies(),
+                karafConfiguration(),
+                karafStandardFeatures(STANDARD),
+                karafSpringFeatures(SPRING),
+                camelFeatures(CAMEL),
+                cxfFeatures(CXF_RT_SECURITY_SAML,
+                        CXF_BINDINGS_SOAP,
+                        CXF_WS_POLICY,
+                        CXF_WS_SECURITY,
+                        CXF_FRONTEND_JAVASCRIPT,
+                        CXF_JAXRS),
+                keystoreAndTruststoreConfig(),
+                contentDirectoryMonitorDependencies(),
+                initContentDirectoryMonitorConfig());
+    }
+
+    private Option testDependencies() {
+        return composite(startBundle("org.awaitility", "awaitility"),
+                startBundle("org.mockito", "mockito-core"),
+                startBundle("org.objenesis", "objenesis"));
+    }
+
+    private Option keystoreAndTruststoreConfig() {
+        InputStream keystore = ContentDirectoryMonitorComponentTest.class.getResourceAsStream(
+                "/serverKeystore.jks");
+        InputStream truststore = ContentDirectoryMonitorComponentTest.class.getResourceAsStream(
+                "/serverTruststore.jks");
+
+        return KeystoreTruststoreConfigurator.createKeystoreAndTruststore(keystore, truststore);
+    }
+
+    private Option initContentDirectoryMonitorConfig() throws IOException {
+        return editConfigurationFilePut(
+                "etc/org.codice.ddf.catalog.content.monitor.ContentDirectoryMonitor-test.cfg",
+                "monitoredDirectoryPath",
+                "");
+    }
+
+    private Option contentDirectoryMonitorDependencies() {
```

Comment:
Even in the case of this, very specific method, with a base class you could yank all the bundle starting knowledge out of the individual component test classes and pass in a list of tuples of group/artifact.